_posts/2023-11-01-#28368.md

```diff
@@ -7,18 +7,73 @@ authors: [ismaelsadeeq]
+ 
+* `removeForBlock` calls the fee estimator [`processBlock`](https://github.com/bitcoin/bitcoin/blob/106ab20f121f14d021725c8a657999079dbabfc1/src/policy/fees.h#L204C11-L204C11) to update the fee stats when transactions are removed from the mempool after a new block connection.
+
+* This `removeForBlock` fee estimator updates block [`ConnectTip`](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/validation.h#L732C13-L732C13). Adding more steps to `processBlock` can slow down block processing even further.
+
+* Subclasses of [`CValidationInterface`](https://github.com/bitcoin/bitcoin/blob/9c30f5ef9d6712539cb8dc0f52b3a2206a25e074/src/validationinterface.h#L77) are called subscribers or clients of `CValidationInterface`. They can implement the callbacks they need and must register with [`CMainSignals`](https://github.com/bitcoin/bitcoin/blob/9c30f5ef9d6712539cb8dc0f52b3a2206a25e074/src/validationinterface.h#L182C7-L182C19) using [`RegisterValidationInterface`](https://github.com/bitcoin/bitcoin/blob/9c30f5ef9d6712539cb8dc0f52b3a2206a25e074/src/validationinterface.h#L26).
+Validation and mempool use [`CMainSignals`](https://github.com/bitcoin/bitcoin/blob/9c30f5ef9d6712539cb8dc0f52b3a2206a25e074/src/validationinterface.h#L182C7-L182C19) to "notify" subscribers of these callbacks.
+The "notification" may entail executing a callback immediately or adding it to a queue of callbacks to be executed in the background. 
+
+  * These callbacks are executed asynchronously on a different thread from the one that fired the event.
+  * `CValidationInterface` has notifications for major events that are happening in the node, when a transaction is added to the mempool [`TransactionAddedToMempool`](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/validationinterface.h#L99), when a transaction is removed from the mempool [`TransactionRemovedFromMempool`](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/validationinterface.h#L133), when a new block is connected to the chain [`BlockConnected`](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/validationinterface.h#L140) and others.
```

Comment:
Also maybe this point should come before the other ones about `CValidationInterface`?
```suggestion
  * `CValidationInterface` includes validation-related events such as addition of a transaction to the mempool [`TransactionAddedToMempool`](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/validationinterface.h#L99), a removal of a transaction from the mempool [`TransactionRemovedFromMempool`](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/validationinterface.h#L133), connection of a block to the chain [`BlockConnected`](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/validationinterface.h#L140), and others.
```